iquip/apps/explorer.py

```diff
@@ -10,6 +10,8 @@
 )
 
 import qiwis
+from iquip.protocols import ExperimentInfo
+from iquip.apps.builder import ExperimentInfoThread
```

Comment:
If it is used here, why did you define `ExperimentInfoThread` in `builder` module?

Reply:
I think that fetching the experiment info is a part of `builder` module becuase it is a step of build.
Of course, we will use the thread in `builder` to edit and update the arguments.

Thus, I added it in `builder` module, not `explorer` module.

Reply:
Hmm.. I think we should keep each app decoupled if possible.
How about passing the full path to the builder constructor and the builder runs the thread to fetch the arguments, not the explorer?

Reply:
Umm... I originally planned as you suggested, but there is a problem.
If we try to open a non-experiment file, we can only check that it is invalid after finishing `ExperimentInfoThread`.
Thus, we implemented as you said, a builder would also be created for an invalid file, and a separate process must be carried out to remove it.

If you're worried about coupling, how about moving the thread to another file such as `iquip/apps/experiment_info_thread.py`?

Reply:
Okay then, but could you explain why `builder` needs the thread? I think all the necessary information is passed when constructing a builder app.

Reply:
I see, then the both apps need the same feature (information fetch).
For this PR, let's go with your solution.
Maybe we can develop the `thread` further to help such thread definitions.